src/main/kotlin/no/nav/helse/sykmelding/domain/Sykmelding.kt

```diff
@@ -21,22 +23,39 @@ data class Sykmelding(val jsonNode: JsonNode) {
     val id = jsonNode["id"].asText()!!
 
     val aktørId = jsonNode["pasientAktoerId"].asText()!!
+
+    val syketilfelleStartDato: LocalDate? = jsonNode["syketilfelleStartDato"].safelyUnwrapDate()
+    val perioder: List<Periode> = jsonNode["perioder"].map {
+        Periode(
+            fom = LocalDate.parse(it["fom"].textValue()),
+            tom = LocalDate.parse(it["tom"].textValue())
+        )
+    }
 }
 
-class SykmeldingSerializer: StdSerializer<Sykmelding>(Sykmelding::class.java) {
+class SykmeldingSerializer : StdSerializer<Sykmelding>(Sykmelding::class.java) {
     override fun serialize(sykmelding: Sykmelding?, gen: JsonGenerator?, provider: SerializerProvider?) {
         gen?.writeObject(sykmelding?.jsonNode)
     }
 }
 
-class SykmeldingDeserializer: StdDeserializer<Sykmelding>(Sykmelding::class.java) {
+class SykmeldingDeserializer : StdDeserializer<Sykmelding>(Sykmelding::class.java) {
     companion object {
         private val objectMapper = jacksonObjectMapper()
-                .registerModule(JavaTimeModule())
-                .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
+            .registerModule(JavaTimeModule())
+            .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
     }
 
     override fun deserialize(parser: JsonParser?, context: DeserializationContext?) =
-            Sykmelding(objectMapper.readTree(parser))
+        Sykmelding(objectMapper.readTree(parser))
 
 }
+
+fun Sykmelding.sykmeldingGjelderFra(): LocalDate? {
+    return listOfNotNull(perioder.map { it.fom }.min(), syketilfelleStartDato).min()
```

Comment:
`minOf(perioder.map { it.fom }.min(), syketilfelleStartDato)`?

Reply:
Kompilatoren sliter litt med nullverdier om jeg skriver det på den måten :(